modules/install/src/main/scala/coursier/install/InstallDir.scala

```diff
@@ -423,36 +423,48 @@ import scala.util.Properties
 
       launcher = actualDest(name)
 
-      sourceAndBytes <- Task.fromEither(
-        InfoFile.readSource(launcher).toRight(new Exception(s"Error reading source from $launcher"))
-      )
-      (source, sourceBytes) = sourceAndBytes
-
-      pathDescriptorBytes <- update(source).flatMap {
-        case Some(res) => Task.point(res)
-        case None => Task.fail(new Exception(s"${source.id} not found in ${source.channel.repr}"))
-      }
-      (path, descriptorBytes) = pathDescriptorBytes
-
-      desc <- Task.fromEither(InfoFile.appDescriptor(path, descriptorBytes))
-
-      appInfo = {
-        val info = AppInfo(desc, descriptorBytes, source, sourceBytes)
-        val foundName = info.appDescriptor.nameOpt
-          .getOrElse(info.source.id)
-        if (foundName == name)
-          info
-        else
-          // just in case, that shouldn't happen
-          info.withAppDescriptor(info.appDescriptor.withNameOpt(Some(name)))
-      }
+      writtenOpt <-
+        if (Files.exists(launcher))
```

Comment:
This is the important change (the content of the “then-branch” is an exact copy of what we had before). Now, before we try to read the source, we check if the file exists. In case that file does not exist, we immediately return `Some(false)`.